ui/src/dashboards/components/DashboardHeaderEdit.js

```diff
@@ -8,28 +8,46 @@ class DashboardEditHeader extends Component {
     const {dashboard: {name}} = props
     this.state = {name}
     this.handleChange = ::this.handleChange
+    this.handleFormSubmit = ::this.handleFormSubmit
+    this.handleKeyUp = ::this.handleKeyUp
   }
 
   handleChange(name) {
     this.setState({name})
   }
 
+  handleFormSubmit(e) {
+    e.preventDefault()
+    const name = e.target.name.value
+    this.props.onSave(name)
+  }
+
+  handleKeyUp(e) {
+    const {dashboard: {name}, onCancel} = this.props
+    if (e.key === 'Escape') {
+      this.setState({name})
+      onCancel()
+    }
+  }
+
   render() {
     const {onSave, onCancel} = this.props
     const {name} = this.state
 
     return (
       <div className="page-header full-width">
         <div className="page-header__container">
-          <div className="page-header__left">
+          <form className="page-header__left" onSubmit={this.handleFormSubmit}>
```

Comment:
I'm not a fan of using `form`s as it brings up the HTML5 tooltip. Either way, we should confirm with @nhaugo and @alexpaxton on the design / usage of HTML5 form validation + tooltips. The tooltips feel out of sync with the design for the rest of the app.